### ISSUE TITLE
Lost some textures when loading .gltf file

### ISSUE THREAD

**Issue by saucebing** [mrdoob/three.js#16469]
I have loaded a .gltf file but found that it lost some texture.
Here is what it supposed to be:
<img width="675" alt="Screenshot 2019-05-17 at 10 13 29" src="https://user-images.githubusercontent.com/4333044/57919123-14236b00-7890-11e9-9194-2b4b2c546807.png">

But it looks like as below after loading:
<img width="1222" alt="Screenshot 2019-05-17 at 10 15 53" src="https://user-images.githubusercontent.com/4333044/57919171-32896680-7890-11e9-8785-cb888bb2c606.png">

We could find that it lost a large parts of texture especially on the wall.

**The .gltf file is here for testing:**
[007.zip](https://github.com/mrdoob/three.js/files/3190880/007.zip)

By the way, for the convenience of your testing, the Box2825 is the one enclosed by turquoise rectangle:
<img width="997" alt="Screenshot 2019-05-17 at 10 28 34" src="https://user-images.githubusercontent.com/4333044/57919579-fa365800-7890-11e9-9745-8d0d201b5609.png">

When I deleted it, it looked like:
<img width="964" alt="Screenshot 2019-05-17 at 10 28 48" src="https://user-images.githubusercontent.com/4333044/57919609-0a4e3780-7891-11e9-9e13-e074bcefe293.png">
 
Maybe you can check why this texture was not loaded correctly at the beginning.

##### Three.js version

- [x] Dev
- [ ] r104
- [ ] ...

##### Browser

- [ ] All of them
- [x] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [ ] All of them
- [ ] Windows
- [x] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

**Comment by Mugen87** [comment]
Can you please try to use PNG textures instead of TGA?

**Comment by saucebing** [comment]
But it has loaded some .gltf files with .tga textures very well so I thought it would be a bug. To be honest, I used 3DS MAX + Babylon plugin to export the .gltf file because the original 3DS MAX could not export .gltf files and the Babylon plugin has no such an export option to choose the type of the texture.

**Comment by saucebing** [comment]
I wonder whether it works if I convert .tga files to .png files and then just change paths in the .gltf file (.gltf file is the ASCII format).

**Comment by Mugen87** [comment]
> But it has loaded some .gltf files with .tga textures very well

That's strange since `TGA` is not natively supported in browsers. In `three.js`, `TGA` textures require the usage of `TGALoader` so you can use them correctly. Hence, the usage of `PNG` is more straightforward.

**Comment by saucebing** [comment]
I have another model with some tga textures exported using the same export plugin but it looked like normal. I would quickly double check the detail to make sure whether the normal textures are the tga file or not.

**Comment by saucebing** [comment]
> In three.js, TGA textures require the usage of TGALoader so you can use them correctly

By the way, why don't you call TGALoader within gltfLoader and other loaders?

**Comment by Mugen87** [comment]
According to the [3DS Max Babylon.js glTF exporter](https://doc.babylonjs.com/resources/3dsmax_to_gltf): 

> glTF 2.0 only supports the following image formats: jpg and png. You are adviced to use those formats for your textures when exporting to glTF.

> Note that the exporter also supports textures with bmp, gif, tga, tif and dds formats. But, those textures will be automatically converted to png/jpg by the exporter to follow glTF specifications.

The usage of TGA is actually not intended.

**Comment by saucebing** [comment]
I found that the exporter converted some of my files to png or jpg but didn't convert some other pics. You could see that there are some files with two same filenames but two different extension name in my zip file. I have tried convert tga files to png format and replace them myself in .gltf file. But it didn't work and had no any effect.

As my another gltf file, it is shown in threejs as below, which are normal:
![Screenshot from 2019-05-17 16-23-56](https://user-images.githubusercontent.com/4333044/57938496-3339f100-78c0-11e9-9750-7452a16a3832.png)

And this one was exported with four tga files and one jpg file. **To make sure I didn't mistake something, I open the .gltf file and saw names of these .tga files.** I convert these four tag files to jpg files and shown as below, you can see that they are loaded correctly because you can find them in the scenery, such as the flag.

![Castle_Atlas](https://user-images.githubusercontent.com/4333044/57938564-5b295480-78c0-11e9-96b6-d64a22d689d0.jpg)
![Flag_Torch_Shed](https://user-images.githubusercontent.com/4333044/57938571-5f557200-78c0-11e9-8261-68902733a084.jpg)
![Foliage](https://user-images.githubusercontent.com/4333044/57938575-61b7cc00-78c0-11e9-9ceb-a9c182d52925.jpg)
![Rocks](https://user-images.githubusercontent.com/4333044/57938576-63818f80-78c0-11e9-91ef-40824bfc77dc.jpg)

So the gltfLoader should be able to load .tga file to some extent. Can you double check it?

**Comment by donmccurdy** [comment]
> So the gltfLoader should be able to load .tga file to some extent. 

No, the glTF format disallows TGA entirely. It should be converted to PNG on JPEG at time of export, and if the exporter isn't doing this but includes the TGA file instead, that's a bug in the exporter. Sounds like the exporter intends to do that conversion but isn't here. /cc @bghgary

**Comment by saucebing** [comment]
If it is a bug for the exporter I would try to report it to the exporter. But it is quite strange that the gltfLoader can load the second model with four tga files. If the gltfLoader can load some tga files it should be able to load all tga files. If someone needs to test my second one model please tell me, I can send the zip file for you to test.

**Comment by donmccurdy** [comment]
It is definitely a bug with the exporter – I'd recommend reporting it there. [Browsers do not even support TGA files](https://stackoverflow.com/questions/6083684/how-can-i-show-a-tga-image-in-a-img-tga). I don't know why one of your models looks OK, but a glTF file referencing a TGA texture is not a correct, and won't be fixed or supported.
